lib/acme/client.rb

```diff
@@ -268,12 +269,18 @@ def connection_for(url:, mode:)
     endpoint = "#{uri.scheme}://#{uri.hostname}:#{uri.port}"
     @connections ||= {}
     @connections[mode] ||= {}
-    @connections[mode][endpoint] ||= new_connection(endpoint: endpoint, mode: mode)
+    @connections[mode][endpoint] ||= new_acme_connection(endpoint: endpoint, mode: mode)
   end
 
-  def new_connection(endpoint:, mode:)
-    Faraday.new(endpoint, **@connection_options) do |configuration|
+  def new_acme_connection(endpoint:, mode:)
+    new_connection(endpoint: endpoint) do |configuration|
       configuration.use Acme::Client::FaradayMiddleware, client: self, mode: mode
+    end
+  end
+
+  def new_connection(endpoint:, &block)
+    Faraday.new(endpoint, **@connection_options) do |configuration|
+      yield(configuration) if block
```

Comment:
No need to capture the block, just `if block_given?`

Reply:
👍